transformer_engine/common/libtransformer_engine.version

```diff
@@ -0,0 +1,4 @@
+{
+	global: *nvte*; *transformer_engine*;
```

Comment:
why do we need `*` before `nvte` or `transformer_engine`? Shouldn't it be `nvte_*` instead (since that is the format of our C APIs)? I'm also not convinced we need to expose anything with `transformer_engine`, could you give examples of the APIs we would want to expose that way?

Reply:
1. The * before `nvte` and `transformer_engine` is needed because some symbols are mangled, for e.g. `_Z14nvte_unpermuteP...`.
2. `transformer_engine` is necessary because we use some functionality such as transformer_engine::getenv in the frameworks, so we need to define the symbol for it to load the framework shared objects.

Given how both `*transformer_engine*` and (to a lesser extent) `*nvte*` patterns are specific to TE, I think it's safe to continue exposing these globally.

Reply:
1. Anything that is mangled is a bug since we advertise C API. In the case of permute it is because there is no `extern`"C"` decorator around those functions. Also we are using `nullptr` instead of `NULL` or `0` - apparently C23 introduced it but I don't think we want to expect people to have such new C standards. @phu0ngng could we have it fixed?
2. I thought we do not do this anymore and instead compile that function in each framework extension separately. @timmoon10 could you comment?